classes/class-gf-securesubmit.php

```diff
@@ -1,950 +1,1597 @@
+            return;
+        }
+        if (!$this->isCC($form)) {
+            return;
+        }
+
+        $feeds = GFAPI::get_feeds(null, $form['id']);
+        $feed  = $feeds[0];
+
+        $cc_field = $this->get_credit_card_field($form);
+
+        $args = ['apiKey'    => $this->getPublicApiKey($feed),
+                 'formId'    => $form['id'],
+                 'ccFieldId' => $cc_field['id'],
+                 'ccPage'    => rgar($cc_field, 'pageNumber'),
+                 'isAjax'    => $is_ajax,];
+
+        $script
+            = 'new
+
+
+
+
+        (' . json_encode($args) . ');';
+        GFFormDisplay::add_init_script($form['id'], 'securesubmit', GFFormDisplay::ON_PAGE_RENDER, $script);
+    }
+
+    /**
+     * @param $content
+     * @param $field
+     * @param $value
+     * @param $lead_id
+     * @param $form_id
+     *
+     * @return string
+     */
+    public
+    function addSecureSubmitInputs($content, $field, $value, $lead_id, $form_id)
+    {
+        if (!$this->has_feed($form_id) || GFFormsModel::get_input_type($field) != 'creditcard') {
+            return $content;
+        }
+
+        if ($this->getSecureSubmitJsResponse()) {
+            $content .= '<input type=\'hidden\' name=\'securesubmit_response\' id=\'gf_securesubmit_response\' value=\'' . rgpost('securesubmit_response') . '\' />';
+        }
+
+        return $content;
+    }
+
+    /**
+     * @param array $validationResult
+     *
+     * @return array
+     */
+    public
+    function maybe_validate($validationResult)
+    {
+        if (!$this->has_feed($validationResult['form']['id'], true)) {
+            return $validationResult;
+        }
+
+        foreach ($validationResult['form']['fields'] as $field) {
+            $currentPage        = GFFormDisplay::get_source_page($validationResult['form']['id']);
+            $fieldOnCurrentPage = $currentPage > 0 && $field['pageNumber'] == $currentPage;
+
+            if (GFFormsModel::get_input_type($field) != 'creditcard' || !$fieldOnCurrentPage) {
+                continue;
+            }
+
+            if ($this->getSecureSubmitJsError() && $this->hasPayment($validationResult)) {
+                $field['failed_validation']  = true;
+                $field['validation_message'] = $this->getSecureSubmitJsError();
+            }
+            else {
+                $field['failed_validation'] = false;
+            }
+
+            break;
+        }
+
+        $validationResult['is_valid'] = true;
+
+        return parent::maybe_validate($validationResult);
+    }
+
+    /**
+     * @param $validation_result
+     *
+     * @return mixed
+     */
+    public
+    function validation($validation_result)
+    {
+        if (!rgar($validation_result['form'],'id',false)) {
+            return $validation_result;
+        }
+        if (!$this->has_feed($validation_result['form']['id'], true)) {
+            return $validation_result;
+        }
+
+        $isACH = false;
+        $isCC = false;
```

Comment:
These begin as bools but become something else. Is false an appropriate starting value?